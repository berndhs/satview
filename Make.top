#
# top level makefile for cloudtrack
#
# //
# //  Copyright (C) 2010 - Bernd H Stramm 
# //
# // This program is distributed under the terms of 
# // the GNU General Public License version 3 
# //
# // This software is distributed in the hope that it will be useful, 
# // but WITHOUT ANY WARRANTY; without even the implied warranty 
# // of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
# //


#QMAKE = `sh which_qmake.sh`
QMAKE = qmake
PROFILE_FILE = cloudtrack.pro
EXEC_VIEW = cloudtrack
EXEC_COPY = copydb
DIR_VIEW = viewer
DIR_COPY = copier

EXEC = $(DIR_VIEW)/$(EXEC_VIEW) $(DIR_COPY)/$(EXEC_COPY)
MAKE_Q = Makefile
MAKEFILE = Make.top

DESTDIR = $(HOME)/bin

all:
	sh build.sh

$(MAKE_Q):
	$(QMAKE) $(PROFILE_FILE)

qt:
	$(QMAKE) $(PROFILE_FILE)

d: 
	$(MAKE) -f $(MAKE_Q) debug

r: 
	$(MAKE) -f $(MAKE_Q) release
  
  
clean:
	if [ -e $(MAKE_Q) ]; then $(MAKE) -i -f $(MAKE_Q) clean; fi 
	if [ -e $(MAKE_Q) ]; then $(RM) $(MAKE_Q)* ; fi 
	$(RM) $(EXEC)
  
install:
	install $(EXEC) $(DESTDIR)
