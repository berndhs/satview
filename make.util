# Makefile for the satview program

# //
# //  Copyright (C) 2009 - Bernd H Stramm
# //
#
# // This program is distributed under the terms of 
# // the GNU General Public License version 3 
# //
# // This software is distributed in the hope that it will be useful, 
# // but WITHOUT ANY WARRANTY; without even the implied warranty 
# // of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
# //



#
#
#
 
INSTALLDIR = $(HOME)/bin

ALLSRC = \
blob-image.cpp \
controlpanel.cpp \
copydb.cpp \
cmdoptions.cpp \
dbconnect.cpp \
frametypes.cpp \
imagewin.cpp \
satpicbuf.cpp \
satpiclist.cpp \
shapeframe.cpp \
sjdatasrc.cpp \
textbox.cpp \
trackmain.cpp \
clouddebug.cpp \
version.cpp 

ALLHDR = \
berndsutil.h \
blob-image.h \
controlpanel.h \
cmdoptions.h \
dbconnect.h \
fault.h \
frametypes.h \
imagewin.h \
satpicbuf.h \
satpiclist.h \
satview-config.h \
satview-defaults.h \
shapeframe.h \
sjdatasrc.h \
textbox.h \
trackmain.h \
clouddebug.h \
version.h

ALLUI = control.ui imagewin.ui getstring.ui

ALLMAKE = Makefile cloud.pro copydb.pro

ALLDOC = Doxyfile userman.txt userman.html INSTALL COPYING

ALLARCHIVE = $(ALLSRC) $(ALLHDR) $(ALLUI) $(ALLMAKE) $(ALLDOC)

TARTARGET = cloudtrack.tar.gz

tar:
	tar zcvf $(TARTARGET) $(ALLARCHIVE)
	@ls -l $(TARTARGET)


install:
	install -p cloudtrack copydb $(INSTALLDIR)

docu:
	doxygen

userman:
	asciidoc  -n userman.txt
	
