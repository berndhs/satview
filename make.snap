
TARGET = cloudtrack
TIMESTAMP=date -u +"%Y%m%d-%H%M%S-u"

SRC = *.cpp
HDR = *.h
IMG = *.png
DOC = LICENSE.txt INSTALL
TOOLS = *.ui *.pro *.qrc ship/*.iss qcopy/*.pro make.snap
ARCHFILES = $(SRC) $(HDR) $(IMG) $(DOC) $(TOOLS)

snap:
	T=$(TARGET).`$(TIMESTAMP)`.tar.gz ; \
	tar zcvf $$T $(ARCHFILES) ; \
	ls -l $$T

arch:
	T=$(TARGET).tar.gz ; \
	tar zcvf $$T $(ARCHFILES) ; \
	ls -l $$T


snapsrc:
	tar zcvf src.tar.gz $(SRC) $(HDR)

